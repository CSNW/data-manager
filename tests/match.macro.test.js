const pluginTester = require('babel-plugin-tester');
const plugin = require('babel-plugin-macros');

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: { filename: __filename },
  tests: [
    `
      import { filter } from 'data-manager';
      import match from '../match.macro';

      filter(match({
        a: 1,
        b: 2,
        c: 3
      }))
    `,
    `
      import { filter } from 'data-manager';
      import match from '../match.macro';

      const min = 0;
      const max = 100;
      const values = [1, 2, 3];

      filter(match({
        // conditions
        a: 1,
        b: { $eq: 2 },
        'c': { $ne: true },
        d: { $gt: 0, $lt: 100},
        e: { $gte: min, $lte: max},
        'f': { $in: [1, 2, 3] },
        g: { $nin: values }
      }))
    `,
    `
      import { filter } from 'data-manager';
      import match from '../match.macro';

      const min = 0;
      const max = 100;
      const values = [1, 2, 3];

      filter(match({
        // logic
        $and: { a: 1, b: 2 },
        $or: { a: { $gt: 1 }, b: { $lt: 1 } },
        $not: { a: 1, b: 2 },
        $nor: { a: { $gt: 1 }, b: { $lt: 1 } }
      }))
    `
  ]
});
