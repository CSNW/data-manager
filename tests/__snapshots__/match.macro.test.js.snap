// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros 1. macros: 1. macros 1`] = `
"
import { filter } from 'data-manager';
import match from '../match.macro';

filter(match({
  a: 1,
  b: 2,
  c: 3
}))

      ↓ ↓ ↓ ↓ ↓ ↓

import { filter } from 'data-manager';
filter(row => {
  return row.a === 1 && row.b === 2 && row.c === 3;
});
"
`;

exports[`macros 2. macros: 2. macros 1`] = `
"
import { filter } from 'data-manager';
import match from '../match.macro';

const min = 0;
const max = 100;
const values = [1, 2, 3];

filter(match({
  // conditions
  a: 1,
  b: { $eq: 2 },
  'c': { $ne: true },
  d: { $gt: 0, $lt: 100},
  e: { $gte: min, $lte: max},
  'f': { $in: [1, 2, 3] },
  g: { $nin: values }
}))

      ↓ ↓ ↓ ↓ ↓ ↓

import { filter } from 'data-manager';
const min = 0;
const max = 100;
const values = [1, 2, 3];
filter(row => {
  return row.a === 1 && row.b === 2 && row['c'] !== true && row.d > 0 && row.d < 100 && row.e >= min && row.e <= max && [1, 2, 3].includes(row['f']) && !values.includes(row.g);
});
"
`;

exports[`macros 3. macros: 3. macros 1`] = `
"
import { filter } from 'data-manager';
import match from '../match.macro';

const min = 0;
const max = 100;
const values = [1, 2, 3];

filter(match({
  // logic
  $and: { a: 1, b: 2 },
  $or: { a: { $gt: 1 }, b: { $lt: 1 } },
  $not: { a: 1, b: 2 },
  $nor: { a: { $gt: 1 }, b: { $lt: 1 } }
}))

      ↓ ↓ ↓ ↓ ↓ ↓

import { filter } from 'data-manager';
const min = 0;
const max = 100;
const values = [1, 2, 3];
filter(row => {
  return row.a === 1 && row.b === 2 && (row.a > 1 || row.b < 1) && !(row.a === 1 && row.b === 2) && !(row.a > 1) && !(row.b < 1);
});
"
`;
